emperor/support_files/js/sceneplotview3d.js

```diff
@@ -100,6 +102,90 @@ define([
     this.dimensionRanges = {'max': [], 'min': []};
     this.drawAxesWithColor(0xFFFFFF);
     this.drawAxesLabelsWithColor(0xFFFFFF);
+
+    var raycaster = new THREE.Raycaster();
+    var mouse = new THREE.Vector2();
+
+    //initialize subscribers for event callbacks
+    this.EVENTS = ['click', 'dblclick'];
+    var subscribers = {};
+    for (var i = 0; i < this.EVENTS.length; i++) {
+      subscribers[this.EVENTS[i]] = [];
+    }
+
+    //create helper function for callbacks
+    var eventCallback = function(eventType, event) {
+      event.preventDefault();
+      //dont do anything if no subscribers
+      if (subscribers[eventType].length === 0) {
+        return;
+      }
+
+      var element = scope.renderer.domElement;
+      mouse.x = ((event.clientX - element.offsetLeft) / element.width) * 2 - 1;
+      mouse.y = -((event.clientY - element.offsetTop) / element.height) * 2 + 1;
+
+      raycaster.setFromCamera(mouse, scope.camera);
```

Comment:
I think that's reasonable. Would it be possible to decompose this logic to simplify the `eventCallback`? What might be nice is something like:

```javascript
 var eventCallback = function(eventType, event) {
    event.preventDefault();
    // don`t do anything if no subscribers
    if (subscribers[eventType].length === 0) {
        return;
    }

    switch(eventType) {
        case 'click':
        case 'dblclick':
            eventMouseSelectPoint();
            break;
        default:
            // should never get here...
            break;
    }
```

It may make sense to do some decomposition on the rest of the event handling system as well given that multiple event types will need to handled. This may be a larger conversation though... but probably painful to do later.

btw, it looks like the events are developer defined given like 152. Can these be part of a controlled vocabulary? It would be useful to make them more descriptive as well, e.g., `MOUSE_LEFT_CLICK` and `MOUSE_LEFT_DOUBLE_CLICK`.